algorithms/unit_tests/TestSortCustomComp.hpp

```diff
@@ -0,0 +1,127 @@
+
+namespace Test {
+namespace SortWithComp {
+
+template <class LayoutTagType, class ValueType>
+auto create_random_view_and_host_clone(
+    LayoutTagType LayoutTag, std::size_t n,
+    Kokkos::pair<ValueType, ValueType> bounds, const std::string& label,
+    std::size_t seedIn = 12371) {
+  using namespace ::Test::stdalgos;
+
+  // construct in memory space associated with default exespace
+  auto dataView = create_view<ValueType>(LayoutTag, n, label);
+
+  // dataView might not deep copyable (e.g. strided layout) so to
+  // randomize it, we make a new view that is for sure deep copyable,
+  // modify it on the host, deep copy to device and then launch
+  // a kernel to copy to dataView
+  auto dataView_dc =
+      create_deep_copyable_compatible_view_with_same_extent(dataView);
+  auto dataView_dc_h = create_mirror_view(Kokkos::HostSpace(), dataView_dc);
+
+  // randomly fill the view
+  Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace> pool(
+      seedIn);
+  Kokkos::fill_random(dataView_dc_h, pool, bounds.first, bounds.second);
+
+  // copy to dataView_dc and then to dataView
+  Kokkos::deep_copy(dataView_dc, dataView_dc_h);
+  // use CTAD
+  CopyFunctor F1(dataView_dc, dataView);
+  Kokkos::parallel_for("copy", dataView.extent(0), F1);
+
+  return std::make_pair(dataView, dataView_dc_h);
+}
+
+template <class T>
+struct MyComp {
+  KOKKOS_FUNCTION
+  bool operator()(T a, T b) const {
+    // we return a>b on purpose here, rather than doing a<b
+    return a > b;
+  }
+};
+
+// clang-format off
+template <class ExecutionSpace, class Tag, class ValueType>
+void run_all_scenarios(int api)
+{
+  using comp_t = MyComp<ValueType>;
+
+  const std::vector<std::size_t> my_scenarios = {0, 1, 2, 9, 1003, 51513};
+  for (std::size_t N : my_scenarios)
+  {
+    auto [dataView, dataViewBeforeOp_h] = create_random_view_and_host_clone(
+        Tag{}, N, Kokkos::pair<ValueType, ValueType>{-1045, 565},
+        "dataView");
+
+    namespace KE = Kokkos::Experimental;
+
+    if (api == 0) {
+      Kokkos::sort(dataView, comp_t{});
+      std::sort(KE::begin(dataViewBeforeOp_h), KE::end(dataViewBeforeOp_h),
+                comp_t{});
+    }
+
+    else if (api == 1) {
+      auto exespace = ExecutionSpace();
+      Kokkos::sort(exespace, dataView, comp_t{});
+      std::sort(KE::begin(dataViewBeforeOp_h), KE::end(dataViewBeforeOp_h),
+                comp_t{});
+      exespace.fence();
+    }
+
+    auto dataView_h = stdalgos::create_host_space_copy(dataView);
+    stdalgos::compare_views(dataViewBeforeOp_h, dataView_h);
+
+    // To actually check that Kokkos::sort used the custom
+    // comparator MyComp, we should have a result in non-ascending order.
+    // We can verify this by running std::is_sorted and if that returns
+    // false, then it means everything ran as expected
+    if (api <= 1 && N >= 2){
+      namespace KE = Kokkos::Experimental;
+      ASSERT_FALSE(std::is_sorted( KE::cbegin(dataView_h), KE::cend(dataView_h)));
```

Comment:
because if we accidentally disregard the comparison functor in the impl, only checking the views would still work. 
I know that "accidentally" here is very fat fetched, but it could be.

Reply:
You mean disregard in our implementation and forget to pass to std::sort when you compute the reference solution?

Reply:
I would drop it but not blocking.